src/wifi/model/wifi-mac.cc

```diff
@@ -494,7 +494,15 @@ WifiMac::ConfigureDcf (Ptr<DcaTxop> dcf, uint32_t cwmin, uint32_t cwmax, bool is
         }
       else
         {
-          dcf->SetTxopLimit (MicroSeconds (1504));
+          if (isDmg)
+            {
+              /* See IEEE802.11-2016 Table 9-137 (Clause 20) */
+              dcf->SetTxopLimit (MicroSeconds (0));
+            }
+          else 
+            {
+              dcf->SetTxopLimit (MicroSeconds (1504));
+            }
```

Comment:
Given the 2016 specs, I would rather do a nested switch/case:
- one on the AC traffic
- one on the type of standard (clause) used
And thus implement all TXOPs for the different clauses